editor/gui/editor_quick_open_dialog.cpp

```diff
@@ -795,6 +765,34 @@ void QuickOpenResultItem::highlight_item(bool p_enabled) {
 	queue_redraw();
 }
 
+void QuickOpenResultItem::draw_search_highlights() {
+	if (dirty_highlights) {
+		// When initially switching layouts, the new sub-item has not yet been positioned, so this
+		// delays finding and drawing highlights until after that happens.
+		dirty_highlights = false;
+		callable_mp(Object::cast_to<CanvasItem>(this), &CanvasItem::queue_redraw).call_deferred();
```

Comment:
`queue_redraw()` is already deferred, so using `call_deferred()` is unnecessary.

Reply:
For some reason this doesn't work so I guess it actually needs to wait twice before the layout values are correct? I'll see about hooking the label's draw instead to improve this stuff